### PR TITLE
Feature: Config option in VSCode Extension Settings

### PR DESCRIPTION
**Pull request by glisom**
Added a setting to pass a file path under .clj-kondo to --config when linting.

### CODE REVIEW

File: server/src/clj_kondo/lsp_server/impl/server.clj [hunk 1]
```diff
@@ -114,14 +115,20 @@
                 (.getParentFile))]
     (config-dir dir)))
 
+(defn config-options
+  [args path]
+  (str/join "/" (concat [path] [(first args)])))
```
Comment (borkdude) [inline server/src/clj_kondo/lsp_server/impl/server.clj]: - [ ] Do we want to support plain EDN config as a string in the VSCode input field as well? This function doesn't account for that.
- [ ] What if the path is absolute?
- [ ] Don't use `str/join` with `"/"` for creating paths, but rather `(io/file ... ...)`, else it won't work on Windows

File: server/src/clj_kondo/lsp_server/impl/server.clj [hunk 2]
```diff
@@ -114,14 +115,20 @@
...
   (let [lang (uri->lang uri)
-        cfg-dir (uri->config-dir uri)
+        ^java.io.File cfg-dir (uri->config-dir uri)
         {:keys [:findings]} (with-in-str text
                               (clj-kondo/run! (cond->
                                                   {:lint ["-"]
                                                    :lang lang}
-                                                cfg-dir (assoc :config-dir cfg-dir))))
+                                                cfg-dir (assoc :config-dir cfg-dir)
+                                                cfg-dir (assoc :config (config-options @arg-options (.getPath cfg-dir))))))
```
Comment (borkdude) [inline server/src/clj_kondo/lsp_server/impl/server.clj]: - [ ] What if there is no config directory and we still want to pass the config to clj-kondo?

File: vscode-extension/package.json [hunk 3]
```diff
@@ -33,5 +33,17 @@
...
     "vscode": "^1.1.35"
+  },
+  "contributes": {
+    "configuration": {
+      "title": "clj-kondo",
+      "properties": {
+        "clj-kondo.config.configArgs": {
+          "type": "string",
+          "scope": "application",
+          "markdownDescription": "Specifies config options to pass to clj-kondo on lint. If specifying a path for an additional edn file, do not include .clj-kondo, that will be included automatically."
```
Comment (borkdude) [inline vscode-extension/package.json:44]: This description isn't really clear on what the input supports: plain EDN, a relative path (resolved against the .clj-kondo dir) and/or absolute paths.

File: vscode-extension/package.json [hunk 4]
```diff
@@ -33,5 +33,17 @@
     "tslint": "^5.16.0",
     "typescript": "^3.6.4",
     "vscode": "^1.1.35"
+  },
+  "contributes": {
+    "configuration": {
+      "title": "clj-kondo",
+      "properties": {
+        "clj-kondo.config.configArgs": {
```
Comment (borkdude) [inline vscode-extension/package.json]: Maybe just call it `clj-kondo.config`, `configArgs` seems a bit redundant to me.

File: server/src/clj_kondo/lsp_server/impl/server.clj (hunk 1 above)
Comment (glisom) [reply server/src/clj_kondo/lsp_server/impl/server.clj]: Will update with io/file and good point I think we want to support absolute or relative. For supporting plain EDN, should I just check to see if the string can be read in by EDN?

File: server/src/clj_kondo/lsp_server/impl/server.clj (hunk 2 above)
Comment (glisom) [reply server/src/clj_kondo/lsp_server/impl/server.clj]: Will make this conditional.

File: vscode-extension/package.json (hunk 3 above)
Comment (glisom) [reply vscode-extension/package.json:44]: I'll clarify along with changes above.

File: server/src/clj_kondo/lsp_server/impl/server.clj [hunk 5]
```diff
@@ -114,14 +116,25 @@
                 (.getParentFile))]
     (config-dir dir)))
 
+(defn config-options
+  [arg ^java.io.File config-dir]
+  (let [file (io/file arg)]
+       (if (.isFile file)
```
Comment (borkdude) [inline server/src/clj_kondo/lsp_server/impl/server.clj:122]: I think this won't work. .isFile only returns true of the file can be resolved against the current working directory.
E.g. 
```
user=> (.isFile (io/file "foobar.edn"))
false
```

A better check may be what we already have in clj-kondo here:

https://github.com/borkdude/clj-kondo/blob/master/src/clj_kondo/impl/core.clj#L66-L67

So if the string doesn't start with `{` or `^`, we assume it must be a file path.

File: server/src/clj_kondo/lsp_server/impl/server.clj [hunk 6]
```diff
@@ -114,14 +116,25 @@
...
+         (if (.isRelative file)
```
Comment (borkdude) [inline server/src/clj_kondo/lsp_server/impl/server.clj:123]: Have you tested this locally? AFAIK Java doesn't have a `isRelative` method. It does have a `isAbsolute` method.

**Comment by borkdude** [comment]
@glisom I thought a little bit more about this and I think we may fix this in a general way so all editors can benefit.

See https://github.com/borkdude/clj-kondo/issues/992. I'd appreciate it if you read it and think this would solve your problem.

**Comment by glisom** [comment]
@borkdude so sorry, got busy at work but can give this a look and get some changes out this weekend!

**Comment by borkdude** [comment]
@glisom If we get #992 right, then I don't think this feature is needed anymore.

**Comment by glisom** [comment]
@borkdude good to close?

**Comment by borkdude** [comment]
@glisom I think so! How is #992 working for you?

**Pull request closed without merging**